app/controllers/users_controller.rb

```diff
@@ -15,9 +15,52 @@ def create
     end
   end
 
+  def all_users
+    if admin?
+      @users = User.all
+      render json: @users, each_serializer: UserSerializer
+    else
+      render json: { error: 'Not Allowed' }, status: :unauthorized
+    end
+  end
```

Comment:
I think here we should return all users except the admin , because he cannot crud him self